tests/test_updater_top_level_update.py

```diff
@@ -424,6 +424,47 @@ def test_new_snapshot_version_rollback(self) -> None:
 
         self._assert_version_equals(Snapshot.type, 2)
 
+    def test_new_snapshot_fast_foward_recovery(self) -> None:
+        """Test snapshot fast-forward recovery using key rotation.
+
+        The snapshot recovery requires the snapshot and timestamp key rotation.
+        It is made by the following steps:
+        - Remove the snapshot and timestamp keys
+        - Create and add a new key for snapshot and timestamp
+        - Rollback snapshot version
+        - Bump and publish root
+        - Bump the timestamp
+        """
+
+        # attacker updates to a higher version (bumping timestamp is required)
+        self.sim.snapshot.version = 99999
+        self.sim.update_timestamp()
+
+        # client refreshes the metadata and see the new snapshot version
+        self._run_refresh()
+        self._assert_version_equals(Snapshot.type, 99999)
+
+        # repo add new snapshot and timestamp keys and recovers snapshot version
+        self.sim.root.roles["snapshot"].keyids.clear()
```

Comment:
I didn't even notice this in the previous PR but it's a bit inconsistent that both `Snapshot.type` and `"snapshot"` are used... could just use Snapshot.type everywhere, or does it look worse like that?

Reply:
Any objection about fixing it for the previous PR on this PR?

Reply:
if it looks fine to you, go ahead